### PR TITLE
refactor: change totalitems from int to long

### PR DESCRIPTION
**Pull request by ArneD**

### CODE REVIEW

File: src/Be.Vlaanderen.Basisregisters.Api/Search/Pagination/PaginationRequest.cs [hunk 1]
```diff
@@ -54,12 +54,12 @@ public class NoPaginationRequest : IPaginationRequest
...
 
-        public PagedQueryable<T> Paginate<T>(SortedQueryable<T> source, Func<IQueryable<T>, int> countFunc = null)
+        public PagedQueryable<T> Paginate<T>(SortedQueryable<T> source, Func<IQueryable<T>, long> countFunc = null)
         {
             var items = source.Items;
-            var limit = countFunc?.Invoke(items) ?? items.Count();
+            var limit = countFunc?.Invoke(items) ?? items.LongCount();
 
-            var paginationInfo = new PaginationInfo(0, limit, limit, 1);
+            var paginationInfo = new PaginationInfo(0, (int) limit, limit, 1);
```
Comment (CumpsD) [inline src/Be.Vlaanderen.Basisregisters.Api/Search/Pagination/PaginationRequest.cs]: Why not change `PaginationInfo` to accept long long?

Comment (ArneD) [reply src/Be.Vlaanderen.Basisregisters.Api/Search/Pagination/PaginationRequest.cs]: Limit (second parameter for paginationInfo) is not the same as Total Items.

Comment (CumpsD) [reply src/Be.Vlaanderen.Basisregisters.Api/Search/Pagination/PaginationRequest.cs]: Ok, but if total items is a long, it is possible a person will enter a limit which exceeds an int too. So we need to change Limit to long too, no?

Comment (ArneD) [reply src/Be.Vlaanderen.Basisregisters.Api/Search/Pagination/PaginationRequest.cs]: `Limit` = `Pagesize`. Which is currently controlled by us with a maximum.
I would not encourage to have a pagesize larger than an `int`.

Comment (CumpsD) [reply src/Be.Vlaanderen.Basisregisters.Api/Search/Pagination/PaginationRequest.cs]: Right.... Mistaken it with offset :p

**Pull request merged**